bolt/users.go

```diff
@@ -38,17 +38,48 @@ func (s *UsersStore) get(ctx context.Context, id uint64) (*chronograf.User, erro
 	return &u, nil
 }
 
+func (s *UsersStore) each(ctx context.Context, fn func(*chronograf.User)) error {
+	return s.client.db.View(func(tx *bolt.Tx) error {
+		return tx.Bucket(UsersBucket).ForEach(func(k, v []byte) error {
+			var user chronograf.User
+			if err := internal.UnmarshalUser(v, &user); err != nil {
+				return err
+			}
+			fn(&user)
+			return nil
+		})
+	})
+}
+
 // Get searches the UsersStore for user with name
-func (s *UsersStore) Get(ctx context.Context, id string) (*chronograf.User, error) {
-	uid, err := strconv.ParseUint(id, 10, 64)
-	if err != nil {
-		return nil, err
+func (s *UsersStore) Get(ctx context.Context, q chronograf.UserQuery) (*chronograf.User, error) {
+	if q.ID != nil {
+		return s.get(ctx, *q.ID)
 	}
-	u, err := s.get(ctx, uid)
-	if err != nil {
-		return nil, err
+
+	if q.Name != nil && q.Provider != nil {
+		var user *chronograf.User
+		err := s.each(ctx, func(u *chronograf.User) {
```

Comment:
i'm not sure this PR solves the "get users more efficiently" issue, since it's still an each -- thoughts?

Reply:
Ah good. point. This is relatively more performant since it's at the Bolt level, but you're probably right. We should probably leave it open.